beancount/parser/booking_full.py

```diff
@@ -570,6 +594,16 @@ def book_reductions(entry, group_postings, balances,
                 balance.is_reduced_by(units)):
                 # This posting is a reduction.
 
+                # If {*} if specified for this posting, re-book inventory at
+                # average cost before matching lots, so the match is more
+                # likely unambiguous.
```

Comment:
Please insert note on the side-effect on 'balance' here, it's totally not obvious.
I'd much rather have the caller apply it TBH

Reply:
Good point. I removed that side-effect and have the caller apply it instead.